### PR TITLE
Generate the mapping m(github_login)=kerberos_id

### PR DESCRIPTION
**Pull request by hongkailiu**
/cc @openshift/test-platform

### CODE REVIEW

**Comment by hongkailiu** [comment]
https://issues.redhat.com/browse/DPTP-1840

File: cmd/ldap-users-from-github-owners-files/main.go [hunk 1]
```diff
@@ -22,30 +22,40 @@ func main() {
 	ldapFile := flag.String("ldap-file", "", "File generated via ldapsearch -LLL -x -h ldap.corp.redhat.com -b ou=users,dc=redhat,dc=com '(rhatSocialURL=GitHub*)' rhatSocialURL uid")
 	repoBaseDir := flag.String("repo-base-dir", "", "base dir for the target repo. Will be used to resolve OWNERS_ALIASES")
 	repoSubdir := flag.String("repo-sub-dir", ".", "Subdir relative to the --repo-base-dir to look for OWNERS files")
+	mappingFile := flag.String("mapping-file", "", "File to store the mapping file m(github_login)=uid")
```
Comment (droslean) [inline cmd/ldap-users-from-github-owners-files/main.go:25]: The description is confusing

Comment (hongkailiu) [reply cmd/ldap-users-from-github-owners-files/main.go]: See if it is better now.

File: cmd/ldap-users-from-github-owners-files/main.go [hunk 2]
```diff
@@ -22,30 +22,41 @@ func main() {
 	ldapFile := flag.String("ldap-file", "", "File generated via ldapsearch -LLL -x -h ldap.corp.redhat.com -b ou=users,dc=redhat,dc=com '(rhatSocialURL=GitHub*)' rhatSocialURL uid")
 	repoBaseDir := flag.String("repo-base-dir", "", "base dir for the target repo. Will be used to resolve OWNERS_ALIASES")
 	repoSubdir := flag.String("repo-sub-dir", ".", "Subdir relative to the --repo-base-dir to look for OWNERS files")
+	mappingFile := flag.String("mapping-file", "", "File to store the mapping m(github_login)=kerberos_id. When this flag is provided, the program exists after generating the mapping file.")
```
Comment (emilvberglind) [inline cmd/ldap-users-from-github-owners-files/main.go:25]:
```suggestion
	mappingFile := flag.String("mapping-file", "", "File used to store the mapping results of m(github_login)=kerberos_id. When this flag is provided, the program exists after generating the mapping file.")
```

File: cmd/ldap-users-from-github-owners-files/main.go [hunk 3]
```diff
@@ -22,30 +22,41 @@ func main() {
...
 		for _, err := range errs {
-			logrus.WithError(err).Error("encountered error trying to resolve owners")
+			logrus.WithError(err).Warn("encountered error trying to parse ldap file")
 		}
 	}
-	serialized, err := json.Marshal(ldapUsers.List())
-	if err != nil {
-		logrus.WithError(err).Fatal("failed to serialize ldap user list")
+
+	if *mappingFile != "" {
```
Comment (emilvberglind) [inline cmd/ldap-users-from-github-owners-files/main.go:35]: Do we want to add a unit test for the scenario where we're generating the mapping file?

Comment (hongkailiu) [reply cmd/ldap-users-from-github-owners-files/main.go:35]: Updated.

**Review by droslean — APPROVED** [review]

**Pull request merged**